src/Sylius/Component/Core/Checker/CommandBasedContextChecker.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Checker;
+
+use Sylius\Component\Core\Context\ProcessContextInterface;
+
+final class CommandBasedContextChecker implements CommandBasedContextCheckerInterface
```

Comment:
```suggestion
final class CLIContextChecker implements CLIContextCheckerInterface
```
I would like to avoid pure `Command` based naming to not make confusion with commands from messenger